shared/actions/login/saga.js

```diff
@@ -209,10 +209,14 @@ function * passthroughResponseSaga ({response}) {
 
 // TODO type this
 type DisplayAndPromptSecretArgs = any
-const displayAndPromptSecretSaga = (onBackSaga) => function * ({params: {phrase}, response}: DisplayAndPromptSecretArgs) {
-  yield put(Creators.setTextCode(phrase))
+const displayAndPromptSecretSaga = (onBackSaga) => function * ({params: {phrase, previousErr}, response}: DisplayAndPromptSecretArgs) {
+  yield put(Creators.setTextCode(phrase, previousErr))
   yield call(generateQRCode)
-  yield put(navigateAppend(['codePage']))
+
+  // If we have an error, we're already on the right page.
+  if (!previousErr) {
```

Comment:
maybe these should be appends but just switching between siblings so we don't need code like this... dunno how big a change that is but if it simplifies things... esp in this flow where you effectively can't go 'back' at all really